### DISCUSSION TITLE
Might be a bug: Result objects fail to pick non string values stored in index

### DISCUSSION THREAD

**Discussion by yjagota** [ravendb/ravendb#17029] [Q&A]
Hello guys,

I encountered something which I think is a bug, but I am posting it as a question as I might be doing something wrong:

Considering the following objects:
```
public class Sales
{
	public string Id { get; set; }
	public string CustomerId { get; set; }
	public string ProductId { get; set; }
	public decimal Quantity { get; set; }
}

public class Product
{
	public string Id { get; set; }
	public string Name { get; set; }
	public decimal Price { get; set; }
	public bool IsNewProduct { get; set; }
}

public class Customer
{
	public string Id { get; set; }
	public string Name { get; set; }
}
```

And the following index:
```
public sealed class SalesIndex : AbstractIndexCreationTask<Sales, SalesIndex.Result>
{
	public class Result
	{
		public string Id { get; set; }
		public string CustomerId { get; set; }
		public string ProductId { get; set; }
		public string ProductName { get; set; }
		public decimal ProductPrice { get; set; }
		public bool ProductIsNew { get; set; }
		public decimal Quantity { get; set; }
	}

	public SalesIndex()
	{
		Map =
			objects =>
				from o in objects
				let product = LoadDocument<Product>(o.ProductId)
				select new
				{
					o.Id,
					o.CustomerId,
					ProductId = product.Id,
					ProductName = product.Name,
					ProductPrice = product.Price,
					ProductIsNew = product.IsNewProduct,
					o.Quantity
				};

		Index(x => x.ProductName, FieldIndexing.No);
		Index(x => x.ProductPrice, FieldIndexing.No);
		Index(x => x.ProductIsNew, FieldIndexing.No);

		Store(x => x.ProductId, FieldStorage.Yes);
		Store(x => x.ProductName, FieldStorage.Yes);
		Store(x => x.ProductPrice, FieldStorage.Yes);
		Store(x => x.ProductIsNew, FieldStorage.Yes);
	}
}
```

The following query will fail:
```
// Note that we are using the 'SalesIndex.Result' object and not the 'Sales' object...
var query = from o in repo.Query<SalesIndex.Result, SalesIndex>()
	where o.CustomerId.In("ABC", "123", "456", "789", "XYZ")
	let customer = RavenQuery.Load<Customer>(o.CustomerId)
	select new
	{
		CustomerName = customer.Name, // This will work...

		o.ProductName, // This will work, even though this is also being picked up from index..
		o.ProductPrice, // This will fail with => Error converting value {null} to type 'System.Decimal'. Path 'ProductPrice'.
		o.ProductIsNew, // This will fail with => Error converting value {null} to type 'System.Boolean'. Path 'ProductIsNew'.

		o.Quantity // This will work...
	};
```

Is this by design? Seems like a error to me because if it is how it is supposed to work, then o.ProductName should also have 'null' values after the query, but it contains the correct data.

**Comment by garayx** [comment]
Hi

What is the actual data that are you using ? Can you submit a complete failing unit test ? 
https://ravendb.net/docs/article-page/5.4/Csharp/start/test-driver#example-1

It seems that you have nulls values in `ProductPrice` & `ProductIsNew` , and you cannot put null into bool & decimal since they are not nullable types.

**Comment by yjagota** [comment]
Found the issue. Actually there were no nulls in the results, but instead there was a field missing in a calculated field, something like:
`Total = x.Price * x.Scheme * x.Discount`, where Discount field was not present in the document. This field was added later in development and there was an old document which did not had that field which was also causing an index error which I did not notice. Although I do not understand why it was causing the boolean to cause the error as well, as it should have only caused error on the decimal field, but adding the field to the document fixed both the errors.

Thanks.
